src/Game/AccountManagement.java

```diff
@@ -1077,6 +1077,55 @@ public static void draw_change_pass_hook(int mouseX, int mouseY, int mouseButton
             "the same as your username", Renderer.width / 2, yPos, 4, 0xFFFFFF);
         return;
       }
+    } else {
+        // TODO: this is bad practice, but I'm putting XPBar code in this file
+        // so that I don't have to reimplement this input box.
+        // If we're going to do this, this entire method should be put in some other file, possibly its own class
+        if (panelPasswordChangeMode == 8) {

```

Comment:
It is very bad practice indeed. I think should be placed to its own space, also in case other people already have used custom password check indices 8 and onwards in their fork. Probably will give you a starting diff to have it placed appropriately without mixing in

Reply:
Something around the lines in this patch [https://cdn.discordapp.com/attachments/449425134734934046/789259377701945354/changes.patch](url) so it gets structured better